statsmodels/genmod/generalized_linear_model.py

```diff
@@ -214,14 +214,40 @@ def __init__(self, endog, exog, family=None, offset=None, exposure=None,
         self._check_inputs(family, self.offset, self.exposure, self.endog)
         if offset is None:
             delattr(self, 'offset')
+            self._init_keys.remove('offset')
         if exposure is None:
             delattr(self, 'exposure')
+            self._init_keys.remove('exposure')
         #things to remove_data
         self._data_attr.extend(['weights', 'pinv_wexog', 'mu', 'data_weights',
                                 '_offset_exposure'])
         # register kwds for __init__, offset and exposure are added by super
         self._init_keys.append('family')
 
+        self.nobs = len(endog)
+
+        # Construct a combined offset/exposure term.  Note that
+        # exposure has already been logged if present.
+        offset_exposure = 0.
+        if hasattr(self, 'offset'):
+            offset_exposure = self.offset
```

Comment:
general comment: those `if hasattr`  can often be replaced by `getattr` with default
`offset_exposure = getattr(self, 'offset', 0)` should work in this